src/Sulu/Bundle/HttpCacheBundle/DependencyInjection/SuluHttpCacheExtension.php

```diff
@@ -0,0 +1,142 @@
+    }
+
+    /**
+     * Configure the proxy client services
+     *
+     * @param array $config
+     * @param ContainerBuilder $container
+     */
+    private function configureProxyClient($config, ContainerBuilder $container)
+    {
+        $this->configureProxyClientVarnish($config['varnish'], $container);
+
+        // get default
+        $proxyClientName = null;
+        foreach ($config as $name => $proxyClient) {
+            if ($proxyClient['enabled'] === false) {
+                continue;
+            }
+
+            if (null !== $proxyClientName) {
+                throw new InvalidConfigurationException(sprintf(
+                    'Cannot enable more than one proxy, trying to enable "%s" when "%s" is already enabled',
+                    $name,
+                    $proxyClientName
+                ));
+            }
+
+            $proxyClientName = $name;
+        }
+
+        if (null === $proxyClientName) {
+            $proxyClientName = 'symfony';
+        }
+
+        $container->setParameter('sulu_http_cache.proxy_client.name', $proxyClientName);
+        $container->setAlias('sulu_http_cache.proxy_client', 'sulu_http_cache.proxy_client.' . $proxyClientName);
+
+    }
+
+    /**
+     * Configure the varnish services
+     *
+     * @param array $config
+     * @param ContainerBuilder $container
+     */
+    private function configureProxyClientVarnish($config, ContainerBuilder $container)
+    {
+        foreach ($config as $key => $value) {
+            $container->setParameter(
+                $this->getAlias() . '.proxy_client.varnish.' . $key,
+                $value
+            );
+        }
+    }
+
+    /**
+     * Configure the structure cache handler services
+     *
+     * @param array $config
+     * @param ContainerBuilder $container
+     */
+    private function configureStructureCacheHandlers($config, ContainerBuilder $container)
+    {
+        // remove handlers which have not been enabled
+        foreach ($config as $handlerName => $handlerConfig) {
+            if (false === $handlerConfig['enabled']) {
+                $container->removeDefinition('sulu_http_cache.handler.' . $handlerName);
+                continue;
+            }
+
+            $enabledHandlers[] = $handlerName;
+        }
+
+        $container->setParameter('sulu_http_cache.handler.public.max_age', $config['public']['max_age']);
+        $container->setParameter('sulu_http_cache.handler.public.shared_max_age', $config['public']['shared_max_age']);
+        $container->setParameter('sulu_http_cache.handler.public.use_page_ttl', $config['public']['use_page_ttl']);
+
+        if ($config['aggregate']['enabled']) {
+            if ($config['aggregate']['handlers']) {
+                $handlers = $config['aggregate']['handlers'];
+            } else {
+                $handlers = array_filter($enabledHandlers, function ($name) {
```

Comment:
Add comment: Automatically add all the *enabled* handlers if no handlers are specified.